cluster/network.go

```diff
@@ -262,30 +262,18 @@ func (c *Cluster) checkKubeAPIPort(ctx context.Context) error {
 func (c *Cluster) deployTCPPortListeners(ctx context.Context, currentCluster *Cluster) error {
 	log.Infof(ctx, "[network] Deploying port listener containers")
 
-	etcdHosts := []*hosts.Host{}
```

Comment:
because we handle this part already after deploying the listener:

- deploy listener on host
- if the address already bind to the host, then skip deploying the listener

it doesn't make sense to only deploy on new hosts added to the cluster, this way we deploy on everything which will mitigate any edge case including the one in issue rancher/rancher#15422